src/backend/migrations/Migrations.h

```diff
@@ -0,0 +1,37 @@
+#ifndef MIGRATION_TEMPLATE_H_INCLUDED
+#define MIGRATION_TEMPLATE_H_INCLUDED
+
+namespace Backend {
+namespace Migrations {
+
+template <typename R>
+class Migration
+{
+private:
```

Comment:
We usually omit this `private:` in classes since default visibility is already `private`.